crypto/fipsmodule/cpucap/internal.h

```diff
@@ -165,22 +165,89 @@ OPENSSL_INLINE int CRYPTO_is_VBMI2_capable(void) {
 extern uint32_t OPENSSL_armcap_P;
 extern uint8_t OPENSSL_cpucap_initialized;
 
+// Normalize some older feature flags to their modern ACLE values.
+// https://developer.arm.com/architectures/system-architectures/software-standards/acle
+#if defined(__ARM_NEON__) && !defined(__ARM_NEON)
+#define __ARM_NEON 1
+#endif
+#if defined(__ARM_FEATURE_CRYPTO)
+#if !defined(__ARM_FEATURE_AES)
+#define __ARM_FEATURE_AES 1
+#endif
+#if !defined(__ARM_FEATURE_SHA2)
+#define __ARM_FEATURE_SHA2 1
+#endif
+#endif
+
 // CRYPTO_is_NEON_capable returns true if the current CPU has a NEON unit.
 // If this is known statically, it is a constant inline function.
 // Otherwise, the capability is checked at runtime by checking the corresponding
 // bit in |OPENSSL_armcap_P|. This is also the same for
 // |CRYPTO_is_ARMv8_AES_capable| and |CRYPTO_is_ARMv8_PMULL_capable|
 // for checking the support for AES and PMULL instructions, respectively.
 OPENSSL_INLINE int CRYPTO_is_NEON_capable(void) {
+#if defined(OPENSSL_STATIC_ARMCAP_NEON)
```

Comment:
My preference would be to align more closely with BoringSSL on this and support these flags build flags. A couple of thoughts on this:
* This change is different than the original change that you linked to in that the logic here does not allow the build environment implicit compiler flags (like `__ARM_NEON`) to affect the function. On this, we would still differ from BoringSSL.
* Allowing explicit flags like `OPENSSL_STATIC_ARMCAP_NEON` to be set could be beneficial to consumers, especially when building for devices where the specific CPU capabilities are known and the additional latency from a `OPENSSL_armcap_P` lookup is unnecessary.

Reply:
I'm ok with bringing the `#if`s back to reduce latency. That said,
- we did let the implicit compiler flags (like `__ARM_NEON`) affect the function output both before the commit above (corresponding to the PR that has the comment) and after it (because it affected `OPENSSL_armcap_P` which we were solely checking in the function. So I think with this change, we're modifying the behaviour. Do we intend that?
- Regarding the latency, in the continuation of the discussion of that comment, I asked about the performance impact and billbo-yang confirmed that he saw no change.